libs/ui/src/export_logs_modal.tsx

```diff
@@ -101,6 +103,19 @@ export function ExportLogsModal({
     assert(window.kiosk);
     setCurrentState(ModalState.Saving);
 
+    if (logFileType === LogFileType.Raw && onExportLogs && !openFileDialog) {
+      const result = await onExportLogs(logFileType);
+      await logger.log(LogEventId.FileSaved, userRole, {
+        disposition: result ? 'success' : 'failure',
+        message: result
+          ? 'Sucessfully saved logs on the usb drive.'
+          : 'Failed to save logs to usb drive.',
+        fileType: 'logs',
+      });
```

Comment:
I've been moving more and more event logging from the frontend to the backend, when possible. The `createLogsApi` function that you've added could take a logger as an input and then handle the logging internally.

My thinking is that every time we call the backend's log-exporting function, we want to emit this log, so might as well colocate the log instead of requiring consumers to remember to log. Also avoids cluttering the frontend.

But I don't feel too strongly about this and am okay with leaving as is!

Reply:
Ah got it! Def feel free to time-box. And a good reminder of another challenge that we need to figure out around backend code sharing patterns